pkg/flow/internal/controller/loader.go

```diff
@@ -268,14 +268,15 @@ func (l *Loader) EvaluateDependencies(parentScope *vm.Scope, c *ComponentNode) {
 // held when calling evaluate.
 func (l *Loader) evaluate(parent *vm.Scope, c *ComponentNode) error {
 	ectx := l.cache.BuildContext(parent)
-	if err := c.Evaluate(ectx); err != nil {
-		level.Error(l.log).Log("msg", "failed to evaluate component", "component", c.NodeID(), "err", err)
-		return err
-	}
+	err := c.Evaluate(ectx)
 	// Always update the cache both the arguments and exports, since both might
-	// change when a component gets re-evaluated.
+	// change when a component gets re-evaluated. We also want to cache the arguments and exports in case of an error
```

Comment:
tiniest nit: end comments in periods please :) I'd prefer to treat comments as if they were public-facing documentation.